tests/test_performance.py

```diff
@@ -0,0 +1,40 @@
+from math import log10
+import time
+import timeit
+
+import pytest
+
+from uncertainties import ufloat
+
+
+def ufloat_sum_benchmark(num):
+    sum(ufloat(1, 1) for _ in range(num)).std_dev
+
+
+def time_ufloat_sum_benchmark(num):
+    reps = int(100000 / num)
+    t = timeit.timeit(
+        lambda: ufloat_sum_benchmark(num),
+        number=reps,
+        timer=time.process_time,
+    )
+    return t / reps
+
+
+def test_complexity():
+    result_dict = {}
+    n_list = (10, 100, 1000, 10000, 100000)
+    for n in n_list:
+        result_dict[n] = time_ufloat_sum_benchmark(n)
+    n0 = n_list[0]
+    t0 = result_dict[n0]
+    for n, t in result_dict.items():
+        if n == n0:
+            continue
+        # Check that the plot of t vs n is linear on a log scale to within 10%
+        assert 0.9 * log10(n / n0) < log10(t / t0) < 1.1 * log10(n / n0)
+
+
+@pytest.mark.benchmark
```

Comment:
you can probably parameterise this test with n_list and have it benchmark each num

Reply:
Nesting the benchmark and parametrize marks works in my local environment, but seems to make the github action hang forever running the test.

Reply:
Ok got it. The problem is I was mixing my internal benchmarking using `timeit` with what codspeed was trying to do. The tool is very nice.

Will cleanup the tests a little bit now that I know what's going on then get this ready for review.